### PR TITLE
docs: Modified the setup instructions

### LINKED ISSUE #48: Write Better Setup Instructions for setting the project.
### Describe the bug

As a developer, I would need the correct setup Instructions to set up the projects locally and start working on their issues. But the current Setup instructions aren't correct as well as incomplete and are specifically targeted at users with Linux as Operating System.

### To Reproduce
1. Go to https://github.com/anitab-org/open-source-programs-backend
2. Scroll down to Setup.
3. See the Instructions.

image

There should be instructions for Windows users to set up the database

image

The cd command should be after the virtual environment commands. Also the steps are incomplete.

### Expected behavior

Clear, Concise and Correct Instructions to set up the project and database for different operating Systems(preferably Windows, Linux, mac)

### Desktop (please complete the following information):
 - OS: Ubuntu 20.04
 - Browser Firefox web Browser
 - Version --

### PR DESCRIPTION
**Pull request by khushishikhu**
### Description

Modified the setup instructions of the project.

Fixes #48

### Type of Change:

- Documentation

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] I have written Kotlin Docs whenever is applicable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
     virtualenv venv
     source venv/bin/activate
     pip install -r requirements.txt
+    ````
+6. To run the migrations run: 
+   ````
+   python manage.py migrate
+   ````
+7. To run the server:
+    ```
```
Comment (isabelcosta) [inline README.md:59]:
```suggestion
    ```
6. To run the migrations run: 
   ```
   python manage.py migrate
   ```
7. To run the server:
    ```
```

File: README.md [hunk 2]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
     cd open-source-programs-backend
+    ````
+5. You don't need to create virtual environment while working in Linux.
+    `````
     virtualenv venv
     source venv/bin/activate
     pip install -r requirements.txt
+    ````
+6. To run the migrations run: 
+   ````
```
Comment (isabelcosta) [inline README.md]: This type of code block markdown usually is 3 "`" instead of 4. Can you change the "````" to "```" please :) throughout the whole README

File: README.md [hunk 3]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
+3. Put that download file in the project's directory.
 
-
-2. To start the server:
+4. Move into the project's directory.
 
     ```
     cd open-source-programs-backend
+    ````
+5. You don't need to create virtual environment while working in Linux.
```
Comment (isabelcosta) [inline README.md]: You can though, I usually do this even on Linux. You can rephrase it to something like:
```
Create virtual environment (this is not a hard requirement, but its advisable)
```
This is so if the contributor has multiple projects with different dependencies, they can be in separate environments.

**Review by isabelcosta — CHANGES_REQUESTED** [review]
I just left some minor change requests, but this is looking good :)
Thank you for improving this document @khushishikhu 🎉

File: README.md [hunk 4]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
     ```
+    cd open-source-programs-backend
     sudo -i -u postgres
     createuser osp --pwprompt
     psql
     CREATE DATABASE osp;
     \c osp;
     GRANT ALL PRIVILEGES ON DATABASE osp to osp;
     ```
+2. you need to download API key file from your user-settings on Zulip. The file you download is named as 'download' or rename that to 'download'.
```
Comment (Aaishpra) [inline README.md]: Capital 'Y' here (starting of new line)

File: README.md [hunk 5]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
-5. To run the migrations run: `python manage.py migrate`
-6. You can terminate the process by `Ctrl+C` in your terminal.
+8. Navigate to `http://localhost:8000/` in your browser.
+9. To change the port you may run `python manage.py runserver 0.0.0.0:<port_number>`
+10. To run the migrations run: `python manage.py migrate`
+11. You can terminate the process by `Ctrl+C` in your terminal.
+
+Follow the given instructions for Login into the app.
+
+1. To create the superuser run:
```
Comment (Aaishpra) [inline README.md]: Although creating a superuser is an easy and fast method to login into the application. The correct way is to register a new user by setting up SendGrid details in the settings.py following the instructions from https://sendgrid.com/docs/for-developers/sending-email/integrating-with-the-smtp-api/ you can list this as the first step so that users can register a new user and then login into the application and if that doesn't work then they can create a superuser to login.

**Comment by codesankalp** [comment]
I think adding a line `Download zulip API Key and place it in parent project directory.` before running `python manage.py runserver`.

I faced this while setting up this project yesterday, so I think it will be worth mentioning here.

**Review by Aaishpra — CHANGES_REQUESTED** [review]
Thanks for the quick PR @khushishikhu I have reviewed it and suggested some changes can you please make them alongside @isabelcosta suggestions.

**Comment by Aaishpra** [comment]
> I think adding a line `Download zulip API Key and place it in parent project directory.` before running `python manage.py runserver`.
> 
> I faced this while setting up this project yesterday, so I think it will be worth mentioning here.

It's mentioned in step 2 @codesankalp

File: README.md [hunk 6]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
+    cd open-source-programs-backend
     sudo -i -u postgres
     createuser osp --pwprompt
     psql
     CREATE DATABASE osp;
     \c osp;
     GRANT ALL PRIVILEGES ON DATABASE osp to osp;
     ```
+2. you need to download API key file from your user-settings on Zulip. The file you download is named as 'download' or rename that to 'download'.
+3. Put that download file in the project's directory.
```
Comment (Aaishpra) [inline README.md]: 'Place' can be a better word than 'Put' I think. 🤔

**Comment by codesankalp** [comment]
Sorry, @Aaishpra I didn't read the files-changed of this pr. I wrote a comment here by reading the original README.

**Comment by Aaishpra** [comment]
> Sorry, @Aaishpra I didn't read the files-changed of this pr. I wrote a comment here by reading the original README.

No worries, Thanks for the suggestion though. 👍🏻

File: README.md [hunk 7]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
+5. Create virtual environment (this is not a hard requirement, but its advisable)
+    ```
     virtualenv venv
     source venv/bin/activate
     pip install -r requirements.txt
+    ```
+6. To run the migrations run: 
+   ```
+   python manage.py migrate
+   ```
```
Comment (devkapilbansal) [inline README.md:57]:
```suggestion
   ``python manage.py migrate``
```

@isabelcosta shouldn't this be changes like this

File: README.md [hunk 8]
```diff
@@ -26,31 +26,51 @@ Next follow these instructions.
...
+4. Move into the project's directory.
 
     ```
     cd open-source-programs-backend
+    ````
+5. You don't need to create virtual environment while working in Linux.
+    `````
     virtualenv venv
     source venv/bin/activate
     pip install -r requirements.txt
```
Comment (Soumi7) [inline README.md:52]: When I did this, it worked only then i wrote ```python3 -m pip install requirements.txt```.

Also, prior to this, I had to install the dependencies for psycopg2, mainly python3-dev and libpq-dev. Then the installation worked for me.

Comment (Aaishpra) [reply README.md:52]: If you're using python 3.8 you can install the requirements just by writing 
``` 
pip3 install - r requirements.txt
```
@Soumi7 no need to install anything else

Comment (Soumi7) [reply README.md:52]: @Aaishpra I see. I was using python 3.6 hence the issues. Okay.

**Comment by isabelcosta** [comment]
@devkapilbansal @Aaishpra can you please review this again?

**Review by Aaishpra — DISMISSED** [review]
Thanks for the changes @khushishikhu great work

File: README.md [hunk 9]
```diff
@@ -26,31 +26,53 @@ Next follow these instructions.
...
-5. To run the migrations run: `python manage.py migrate`
-6. You can terminate the process by `Ctrl+C` in your terminal.
+8. Navigate to `http://localhost:8000/` in your browser.
+9. To change the port you may run `python manage.py runserver 0.0.0.0:<port_number>`
+10. To run the migrations run: `python manage.py migrate`
+11. You can terminate the process by `Ctrl+C` in your terminal.
+
+Follow the given instructions for Login into the app.
+
+1. You can register a new user by setting up SendGrid details in the settings.py , you can take a refrence from [here](https://sendgrid.com/docs/for-developers/sending-    email/integrating-with-the-smtp-api/)
```
Comment (isabelcosta) [inline README.md]:
```suggestion
1. You can register a new user by setting up SendGrid details in the `settings.py`, you can take a reference from [here](https://sendgrid.com/docs/for-developers/sending-email/integrating-with-the-smtp-api/)
```

File: README.md [hunk 10]
```diff
@@ -26,31 +26,53 @@ Next follow these instructions.
...
+
+2. To create the superuser run:
+   ```
+   python manage.py createsuperuser
+   ````
+   Fill up the things it asked to and then Login into the app.
 
 ## Contributing
 Please read the Contibuting guidelines, [Code of Conduct](https://github.com/anitab-org/open-source-programs-backend/blob/develop/CODE_OF_CONDUCT.md) and [Reporting Guidelines](https://github.com/anitab-org/open-source-programs-backend/blob/develop/REPORTING_GUIDELINES.md)
```
Comment (codesankalp) [inline README.md:78]:
```suggestion

Please read the [Contributing guidelines](https://github.com/anitab-org/open-source-programs-backend/blob/develop/.github/CONTRIBUTING.md), [Code of Conduct](https://github.com/anitab-org/open-source-programs-backend/blob/develop/CODE_OF_CONDUCT.md) and [Reporting Guidelines](https://github.com/anitab-org/open-source-programs-backend/blob/develop/REPORTING_GUIDELINES.md)
```
Please note there is a spelling mistake in contributing.

**Review by codesankalp — COMMENTED** [review]
Please change this in this instead of creating a new PR.

**Comment by khushishikhu** [comment]
> Please change this in this instead of creating a new PR.

Actually this issue was for writing the setup instructions. That's why I have created a new issue

File: README.md (hunk 7 above)
Comment (isabelcosta) [reply README.md:57]: why @devkapilbansal ? I am not sure what "``" does 🤔 I am really curious
I think like it is, its ok

File: README.md (hunk 10 above)
Comment (isabelcosta) [reply README.md:78]: can be done on the other PR. So you can forget this bit @khushishikhu

**Review by isabelcosta — APPROVED** [review]
Great work @khushishikhu !

**Comment by isabelcosta** [comment]
@Aaishpra @codesankalp @devkapilbansal can you please review this again?

File: README.md (hunk 7 above)
Comment (devkapilbansal) [reply README.md:57]: It is generally used in multiline codes. For single line code, single backquote is used as per conventions.

**Review by codesankalp — APPROVED** [review]
Reviewed, LGTM :tada:

**Comment by isabelcosta** [comment]
thank you so much @codesankalp for reviewing!
